core/txpool/txpool.go

```diff
@@ -273,11 +286,26 @@ func (p *TxPool) Add(txs []*types.Transaction, local bool, sync bool) []error {
 	// so we can piece back the returned errors into the original order.
 	txsets := make([][]*types.Transaction, len(p.subpools))
 	splits := make([]int, len(txs))
+	errs := make([]error, len(txs))
 
 	for i, tx := range txs {
 		// Mark this transaction belonging to no-subpool
 		splits[i] = -1
 
+		// Check against the tx policy
+		status, err := p.optimismPolicy.ValidateTx(tx)
```

Comment:
We should use a timeout context here although I am not sure what a sane value is, perhaps can start lenient with like 2 seconds or something?